packit_service/worker/checker/copr.py

```diff
@@ -74,6 +75,28 @@ def pre_check(self) -> bool:
         return False
 
 
+class IsPackageMatchingJobView(Checker, GetCoprBuildEventMixin):
+    """
+    When running builds for multiple packages (in monorepo) in one job
+    config, we need to check whether the package that we are handling matches
+    the job configuration.
+    """
+
+    def pre_check(self) -> bool:
+        if (
+            not self.job_config.package
+            or self.copr_event.pkg == self.job_config.package
```

Comment:
is this the correct way to check this @lachmanfrantisek @majamassarini ? does the configuration key need to always be equal to the package built in Copr?

I was also thinking about storing the value in DB for the builds and comparing it there

Reply:
Don't we already save this to the database?
https://github.com/packit/packit-service/blob/716a2a3952dadabd2961756f41e633b8e61817ae/packit_service/models.py#L1401

Reply:
but these are stored after a successful build, and we need to compare the values when reporting the state of the build

Reply:
> is this the correct way to check this @lachmanfrantisek @majamassarini ? does the configuration key need to always be equal to the package built in Copr?

If I am following the idea this check should be needed to solve #2058. 
But I am not really sure this implementation will solve #2058, this check is ok, but we don't use anymore the identifiers (to solve #2055) and in this way even though we have this check we will end up updating the status for the same job as many times as are the packages in the job configuration. 

But I could be wrong, I have just this fear.

Reply:
this PR removes the usage of the monorepo identifiers only from the default Copr project name. And for this solution to work, this check needed to be added (not only for #2058), as now the Copr project name will not differ for different packages.

So, do I understand the monorepo support code correctly, for the one job config like [this](https://github.com/packit/packit-service/issues/2055#issuecomment-1552920077), we create for each package a jobconfigview and resolve the jobs for each? If yes, that should work now with the introduced pre-check, as we will for each jobconfigview check whether the package from Copr matches the package in the jobconfigview.

Reply:
Yes, the monorepo code will create one jobconfigview for every package. And the monorepo code customize the `identifier` value with the name of the package for which the job was built. I don't see a problem with this check but I believe that if we don't use the `identifier` we have 3 different jobs (one copr build job for every package in the list) with just one copr project repo name (and this is ok - is what we want in #2055) and just one status check (and this is not ok - we are making #2058 fail in a new way).

Reply:
Thanks for clarification!

> and just one status check 

what do you mean by this? Since the identifier was not removed anywhere else (only in the Copr project name), it will still be used when reporting.

Reply:
Ok, then I am wrong. I thought that the `identifier` changes should also affect the reporting.